CHANGELOG.next.asciidoc

```diff
@@ -21,6 +21,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Remove the deprecated `xpack.monitoring.*` settings. Going forward only `monitoring.*` settings may be used. {issue}9424[9424] {pull}18608[18608]
 - Added `certificate` TLS verification mode to ignore server name mismatch. {issue}12283[12283] {pull}20293[20293]
 - Autodiscover doesn't generate any configuration when a variable is missing. Previously it generated an incomplete configuration. {pull}20898[20898]
+- `AddProcessMetadata` processor enrich process information with owner name. {issue}21068[21068] {pull}21111[21111]
```

Comment:
@adriansr done

Reply:
What do you think is the proper field name for process owner? The options as I can see are:
`process.username` 
`process.owner`
`process.owner_name`

Or `process.owner` should be an object with it's own fields?
`process.owner.uid`
`process.owner.name`
...

Reply:
@adriansr ?

Reply:
Hi @dddpaul, we're still trying to figure out what makes more sense for this use case.

At some point, `process.user.*` made it into ECS, but then it was reverted. See [this comment.](https://github.com/elastic/ecs/issues/1380#issuecomment-829374848)
So to align with ECS, we could do two things:
1. Add user information to the top-level `user` object.
2. Add a custom object under `process`. For example `owner`.

The first makes more sense, but at the same time is more risky for a processor, as this will cause the processor execution to fail if the input document already contains a `user.name` or `user.id`. If `overwrite_keys` is set, then it will silently overwrite those keys, potentially breaking existing pipelines.

Reply:
As a general purpose processor I think it lacks the context to know if setting a `^user` is the correct thing to do. So setting a key(s) under `process` is the safest thing to do and what I would most expect as a user.

Another alternative would be to offer an addition `target_owner` config option that indicates where (or if) process owner info is added to the event. The [registered_domain](https://www.elastic.co/guide/en/beats/filebeat/7.12/processor-registered-domain.html#processor-registered-domain) and [translate_sid](https://www.elastic.co/guide/en/beats/filebeat/7.12/processor-translate-sid.html) processors take an approach like this.